Jenkinsfile

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env groovy
+
+/**
+ * Build Contiki Applications
+ *
+ */
+
+
+def globals = [XC32_URL: 'http://ww1.microchip.com/downloads/en/DeviceDoc/xc32-v1.42-full-install-linux-installer.run',
+               XC32_PATH: '/opt/microchip/xc32/v1.42/bin',
+               XC32_PATH: '/opt/microchip/xc32/v1.42/bin',
+               CONTIKI: '$HOME/test contiki',
+               CONTIKI_REPO: 'https://github.com/CreatorDev/contiki.git'
+               ]
+
+
+    node('docker && imgtec') {  // Only run on internal slaves as build takes a lot of resources
+    def docker_image
+
+    stage('Prepare Docker container') {
+        // Setup a local docker container to run build on this slave
+        docker_image = docker.image "imgtec/creator-builder:latest" // TODO for now have manually setup on slave
+    }
+
+    docker_image.inside("-v ${WORKSPACE}/../../reference-repos:${WORKSPACE}/../../reference-repos:ro") {
+        stage('Configure') {
+            // Checkout a clean version of the repo using reference repo to save bandwidth/time
+            checkout([$class: 'GitSCM',
+                userRemoteConfigs: scm.userRemoteConfigs,
+                branches: scm.branches,
+                doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
+                submoduleCfg: scm.submoduleCfg,
+                browser: scm.browser,
+                gitTool: scm.gitTool,
+                extensions: scm.extensions + [
+                    [$class: 'CleanCheckout'],
+                    [$class: 'PruneStaleBranch'],
+                    [$class: 'CloneOption', honorRefspec: true, reference: "${WORKSPACE}/../../reference-repos/openwrt.git"],
+                ],
+            ])
+
+            // Before Install
+            sh 'sudo dpkg --add-architecture i386'
```

Comment:
Your using docker for your env. You should not be installing anything. I believe the toolchains are already installed in the docker build image?